test/saturation_adjustment.jl

```diff
@@ -6,12 +6,83 @@ using Test
 using Breeze.Thermodynamics:
     MoistureMassFractions,
     PotentialTemperatureState,
+    MoistStaticEnergyState,
     exner_function,
     density,
     with_moisture,
-    saturation_specific_humidity
+    saturation_specific_humidity,
+    mixture_heat_capacity
 
-using Breeze.MoistAirBuoyancies: temperature
+using Breeze.MoistAirBuoyancies: compute_boussinesq_adjustment_temperature
+using Breeze.Microphysics:
+    WarmPhaseSaturationAdjustment,
+    compute_temperature,
+    adjustment_saturation_specific_humidity
+
+@testset "Saturation adjustment (Microphysics + MoistStaticEnergyState)" begin
+    for FT in (Float32, Float64)
+        @info "Testing saturation adjustment for $FT..."
+        grid = RectilinearGrid(default_arch, FT; size=(1, 1, 1), x=(0, 1), y=(0, 1), z=(0, 1))
+        thermo = ThermodynamicConstants(FT)
+        reference_state = ReferenceState(grid, thermo; base_pressure=101325, potential_temperature=288)
+
+        tol = FT(1e-3)
+        microphysics = WarmPhaseSaturationAdjustment(tolerance=tol)
+
+        # Sample a single cell
+        pᵣ = @allowscalar first(reference_state.pressure)
+        g = thermo.gravitational_acceleration
+        z = FT(1/2)
+
+        # First test: absolute zero
+        q₀ = MoistureMassFractions(zero(FT), zero(FT), zero(FT))
+        𝒰₀ = MoistStaticEnergyState(zero(FT), q₀, z, pᵣ)
+        @test compute_temperature(𝒰₀, microphysics, thermo) == 0
+
+        # Second unsaturated test: choose T, pick qᵗ well below saturation
+        T₁ = FT(300)
+        ρ₁ = density(pᵣ, T₁, q₀, thermo)
+        qᵛ⁺ = saturation_specific_humidity(T₁, ρ₁, thermo, thermo.liquid)
+        qᵗ = qᵛ⁺ / 2 # comfortably unsaturated
+
+        q₁ = MoistureMassFractions(qᵗ, zero(FT), zero(FT))
+        cᵖᵐ = mixture_heat_capacity(q₁, thermo)
+        e₁ = cᵖᵐ * T₁ + g * z #  + ℒ₀ * qᵗ
+        𝒰₁ = MoistStaticEnergyState(e₁, q₁, z, pᵣ)
+
+        @test compute_temperature(𝒰₁, microphysics, thermo) ≈ T₁ atol=sqrt(tol)
+        @test compute_temperature(𝒰₁, nothing, thermo) ≈ T₁ atol=sqrt(tol)
+
+        # Third saturated test: choose T, pick qᵗ well above saturation
+        for T₂ in 270:1:320
+            for qᵗ₂ in 1e-2:1e-3:5e-2
+                T₂ = convert(FT, T₂)
+                qᵗ₂ = convert(FT, qᵗ₂)
+
+                qᵛ⁺₂ = adjustment_saturation_specific_humidity(T₂, pᵣ, qᵗ₂, thermo)
+
+                if qᵗ₂ > qᵛ⁺₂ # saturated conditions
+                    qˡ₂ = qᵗ₂ - qᵛ⁺₂
+                    q₂ = MoistureMassFractions(qᵛ⁺₂, qˡ₂, zero(FT))
+                    cᵖᵐ = mixture_heat_capacity(q₂, thermo)
+                    ℒˡᵣ = thermo.liquid.reference_latent_heat
+                    e₂ = cᵖᵐ * T₂ + g * z - ℒˡᵣ * qˡ₂
+
+                    𝒰₂ = MoistStaticEnergyState(e₂, q₂, z, pᵣ)
+                    @test compute_temperature(𝒰₂, microphysics, thermo) ≈ T₂ atol=sqrt(tol)
```

Comment:
This test is wreaking havoc

<img width="778" height="284" alt="Image" src="https://github.com/user-attachments/assets/4eedd1b8-8392-494a-ab3c-361060278d68" />

Reply:
ahahah yeah. i will tweak it.